ethcore/wasm/src/runtime.rs

```diff
@@ -218,11 +218,21 @@ impl<'a> Runtime<'a> {
 		let key = self.h256_at(args.nth(0)?)?;
 		let val_ptr: u32 = args.nth(1)?;
 
-		self.adjusted_charge(|schedule| schedule.sstore_set_gas as u64)?;
-
 		let val = self.h256_at(val_ptr)?;
+		let former_val = self.ext.storage_at(&key).map_err(|_| Error::StorageUpdateError)?;
+
+		if former_val == H256::zero() && val != H256::zero() {
+			self.adjusted_charge(|schedule| schedule.sstore_set_gas as u64)?;
+		} else {
+			self.adjusted_charge(|schedule| schedule.sstore_reset_gas as u64)?;
```

Comment:
So the decision table for this check looks like this:

case |  rate     |
------------ | -------------
0 → 0  |     reset |
0 → ≠0 |     set
≠0 → 0 |     reset
≠0 → ≠0   | reset

Is that intended?

Reply:
Yes

Reply:
!= 0 -> 0

also leads to `storage_clears ++` which is later handled by inter-vm rules

Reply:
Ah, I see, thanks.